code/Migration.php

```diff
@@ -14,4 +14,189 @@
+	 * Returns false if the table or any of the columns do not exist.
+	 * Returns true if the SQL query was executed.
+	 *
+	 * @param string $table
+	 * @param array $columns
+	 * @return boolean
+	 */
+	protected static function dropColumnsFromTable($table, array $columns) {
+		$queried = false;
+		$existingColumns = self::getTableColumns($table);
+		if ($existingColumns) {
+			foreach ($columns as $column) {
+				if (array_key_exists($column, $existingColumns)) {
+					DB::query("ALTER TABLE $table DROP COLUMN $column;");
+					$queried = true;
+				}
+			}
+		}
+		return $queried;
+	}
+
+	/**
+	 * Add columns to a database table if they don't exist.
+	 * Returns false if the table does not exist.
+	 * Returns true if the SQL query was executed.
+	 *
+	 * @param string $table
+	 * @param array $columns e.g. ['MyColumn' => 'VARCHAR(255) CHARACTER SET utf8']
+	 * @return boolean
+	 */
+	protected static function addColumnsToTable($table, array $columns) {
+		$queried = false;
+		$existingColumns = self::getTableColumns($table);
+		if ($existingColumns) {
+			foreach ($columns as $column => $properties) {
+				if (!array_key_exists($column, $existingColumns)) {
+					DB::query("ALTER TABLE $table ADD $column $properties;");
+					$queried = true;
+				}
+			}
+		}
+		return $queried;
+	}
+
+	/**
+	 * Gets the value for a single column in a row from the database by the ID column.
+	 * Useful when a field has been removed from the class' `$db` property,
+	 * and therefore is no longer accessible through the ORM.
+	 * Returns `null` if the table, column or row does not exist.
+	 *
+	 * @param string $table
+	 * @param string $columns
+	 * @param string||int $id
+	 * @return string
+	 */
+	protected static function getRowValueFromTable($table, $field, $id) {
+		$value = null;
+		if (self::tableColumnExists($table, $field)) {
+			$query = new SQLQuery();
+			$query->setFrom($table)->setSelect([$field])->setWhere("ID = $id");
+			$results = $query->execute();
+			if ($results) {
+				foreach ($results as $result) {
+					$value = $result[$field];
+					break;
+				}
+			}
+		}
+		return $value;
+	}
+
+	/**
+	 * Gets the values for multiple rows on a database table by the ID column.
+	 * Useful when fields have been removed from the class' `$db` property,
+	 * and therefore are no longer accessible through the ORM.
+	 * Returns an empty array if the table, any of the columns or the row do not exist.
+	 *
+	 * @param string $table
+	 * @param array $columns
+	 * @param string||int $id
+	 * @return array ['FieldName' => value]
+	 */
+	protected static function getRowValuesFromTable($table, array $fields, $id) {
+		$values = [];
+		if (self::tableColumnsExist($table, $fields)) {
+			$query = new SQLQuery();
+			$query->setFrom($table)->setSelect($fields)->setWhere("ID = $id");
+			$results = $query->execute();
+			if ($results) {
+				foreach ($results as $result) {
+					foreach ($fields as $field) {
+						$values[$field] = $result[$field];
```

Comment:
I really wish SS supported (and we ran) PHP 5.5 so we could just use array_column() because this is tedious, being stuck in the past (supporting 5.3).